handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1061,8 +1061,7 @@ private SSLEngineResult wrap(ByteBufAllocator alloc, SSLEngine engine, ByteBuf i
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         boolean handshakeFailed = handshakePromise.cause() != null;
-
-        ClosedChannelException exception = new ClosedChannelException();
+        SSLHandshakeException exception = new SSLHandshakeException("Remote host closed connection during handshake");
```

Comment:
I think we still need to ensure we only create a `SSLHandshakeException` if we are really still handshaking.
```suggestion
        final Exception exception;
        if (handshakePromise.isDone()) {
            exception = new ClosedChannelException();
        } else {
            // Closed before the handshake was done.
           exception = new SSLHandshakeException("Connection closed during handshake");
           exception.initCause(new ClosedChannelException()); 
        }
```